src/te/schedule/schedule_dataflow_rewrite.cc

```diff
@@ -138,6 +138,15 @@ Tensor Schedule::cache_read(const Tensor& tensor, const std::string& scope,
   }
   os << "." << scope;
 
+  // when a schedule has multiple cache_read on the same tensor,
+  // we make sure their op names are unique. e.g., w.shared, w.shared.d, w.shared.d.d
+  for (auto pair : (*this)->stage_map) {
+    auto stage = pair.second;
+    if (stage->op->name == os.str()) {
+      os << ".d";
```

Comment:
Can we add a global map here and mark these name as "w.shared.0", "w.shared.1"?

Reply:
I was thinking about this too, but in this way we need to maintain another set in Schedule. Consider this case doesn't happen frequently now, the current solution should be sufficient. We can definitely use the approach you suggested in the future after naming conflict becomes much common.

Reply:
I changed the naming to `w.shared`, `w_d.shared`, because we use `StrEndsWith(op->name, ".shared")` in thread binding.